src/Uno.UI/UI/Xaml/Controls/Primitives/ButtonBase/ButtonBase.mux.cs

```diff
@@ -0,0 +1,737 @@
+		{
+			// Cache the last pointer position
+			var pointerPoint = args.GetCurrentPoint(this);
+			_pointerPosition = pointerPoint.Position;
+
+			// Determine if the button is still pressed near the pointer's position			
+			if (_isPointerLeftButtonDown &&
+				IsEnabled &&
+				ClickMode != ClickMode.Hover &&
+				_isPointerCaptured &&
+				!_isSpaceOrEnterKeyDown && !_isNavigationAcceptOrGamepadAKeyDown)
+			{
+				var isValid = IsValidPointerPosition();
+				IsPressed = isValid;
+			}
+
+			base.OnPointerMoved(args);
+		}
+
+		/// <summary>
+		/// Capture the pointer.
+		/// </summary>
+		/// <param name="pointer">Pointer.</param>
+		private void CapturePointerInternal(Pointer pointer)
+		{
+			if (!_isPointerCaptured)
+			{
+				_isPointerCaptured = CapturePointer(pointer);
+			}
+		}
+
+		/// <summary>
+		/// PointerPressed event handler.
+		/// </summary>
+		/// <param name="args">Event args.</param>
+		protected override void OnPointerPressed(PointerRoutedEventArgs args)
+		{
+			if (args.Handled)
+			{
+				Cleanup();
+				return;
+			}
+
+			var pointerPoint = args.GetCurrentPoint(this);
+			var pointerProperties = pointerPoint.Properties;
+			var isLeftButtonPressed = pointerProperties.IsLeftButtonPressed;
+
+			if (isLeftButtonPressed)
+			{
+				_isPointerLeftButtonDown = true;
+
+				// Note: even if ClickMode is Press, we capture the pointer and handle the Release args, but we do nothing if Hover
+
+				// Capturing the Pointer ensures that we will be the first element to receive the pointer released event
+				// It will also ensure that if we scroll while pressing the button, as the capture will be lost, we won't raise Click.
+
+				if (!IsEnabled || ClickMode == ClickMode.Hover)
+				{
+					Cleanup();
+					return;
+				}
+
+				args.Handled = true;
+
+				using (var suspender = new StateChangeSuspender(this))
+				{
+					Focus(FocusState.Pointer);
+
+					CapturePointerInternal(args.Pointer);
+					if (_isPointerCaptured)
+					{
+						IsPressed = true;
+					}
+				}
+
+				if (ClickMode == ClickMode.Press)
+				{
+					RaiseClick(args);
+				}
+			}
+
+			Cleanup();
+			void Cleanup()
+			{
+				base.OnPointerPressed(args);
+			}
+		}
+
+		/// <inheritdoc />
+		protected override void OnPointerReleased(PointerRoutedEventArgs args)
+		{
+			if (args.Handled)
+			{
+				Cleanup();
+				return;
+			}
+
+			// This should be changed to Pointer down and only occur when pointer is down or mouse left button is down.
+			_isPointerLeftButtonDown = false;
+
+			if (!IsEnabled || ClickMode == ClickMode.Hover)
+			{
+				Cleanup();
+				return;
+			}
+
+			// We can not put all our logic for this event inside an if block conditioning to this BOOLEAN, as we need to
+			// release our pointer capture and set is pressed to false depending solely on m_bIsSpaceOrEnterKeyDown
+			_shouldPerformActions = (IsPressed && !_isSpaceOrEnterKeyDown && !_isNavigationAcceptOrGamepadAKeyDown);
```

Comment:
I assume this condition is coming from WinUI source code ... but it's surprising that we are not validating `_isPointerCaptured` here

Reply:
Indeed, the capture is not checked there... Not sure why 😀